pkg/securityscan/scanHandler.go

```diff
@@ -119,6 +104,23 @@ func (c *Controller) handleClusterScans(ctx context.Context) error {
 				}
 				objects = append(objects, cisjob.New(obj, profile, c.Name, c.ImageConfig), cmMap["configcm"], cmMap["plugincm"], cmMap["skipConfigcm"], service)
 
+				if obj.Spec.ScheduledScanConfig != nil && obj.Spec.ScheduledScanConfig.ScanAlertRule != nil {
+					if obj.Spec.ScheduledScanConfig.ScanAlertRule.AlertOnComplete || obj.Spec.ScheduledScanConfig.ScanAlertRule.AlertOnFailure {
```

Comment:
obj.Spec.ScheduledScanConfig.ScanAlertRule was being sent by UI even if user did not select any alerts and this led to creating a PrometheusRule without any rules. So now we check specifically that one of the AlertOnComplete/AlertOnFailure is selected before creating the PrometheusRule.